ctk/features/switch.feature

```diff
@@ -0,0 +1,137 @@
+Feature: Switch Task
+  As an implementer of the workflow DSL
+  I want to ensure that the Switch task behaves correctly
+  So that my implementation conforms to the expected behavior
+
+  Scenario: Switch task with matching case
+    Given a workflow with definition:
+    """yaml
+    document:
+      dsl: 1.0.0-alpha1
+      namespace: default
+      name: switch-match
+    do:
+      switchColor:
```

Comment:
I do not see an if-else  construct, although it can be emulated by
```
switch : 
    when: '.color == "red"'
    then: redAlert
    when: 'color != "red"'
    then: noWorries
```
I feel it would be better to do something like (which also cover the swith case)
condition:
      if: 'color=="red"'
     then: redAlert
     elseif: 'color=="green"'
      then: noWorries
     else: 
      then: willBeGoodToTakeALook

Reply:
I did that for the sake of (JsonPointers) references, as it's easier to reference a named case than an indexed one.

Reply:
Im not sure conditions should be referenced (task I agree they should). Anyway I think the most important part of my comment is the altarnative syntax, which is a superset of the current one.  if you prefer to keep the current one, I think adding a `default:` (equivalent to `else`) to the current switch will do

Reply:
I personnally prefer the current state, that I find clearer, but that's just that, personnal preferences.

As for the `default` state, it exists. At most one case can not define a `when`. You can name it however you want, though `default` should be prefered for clarity.

Not defining a default ("whenless") case is the same than saying you want to do the parent switch task's `then`.

This is documented by the scenarii `Switch task with implicit default case` and `Switch task with explicit default case` of the `switch.feature`

Reply:
ok, an extra condition with no when

Reply:
Yes, exactly!

Reply:
yes resolved